EpikCord/guild.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime
+
+from discord_typings import GuildMemberData, RoleData, RoleTagsData
+
+from .client import Client
+from .flags import GuildMemberFlags, Permissions
+from .user import User
+from .utils import instance_or_none, int_or_none
+
+
+class GuildMember:
+    def __init__(self, client: Client, data: GuildMemberData):
+        self.client = client
+        self.user = instance_or_none(
+            User, data.get("user"), client, data.get("user"), ignore_value=True
+        )
+        self.nick = data.get("nick")
+        self.avatar = data.get("avatar")
+        # self.roles = [Role(client, role) for role in data.get("roles", [])]
+        self.roles = data["roles"]
+        self.joined_at = (
+            datetime.fromisoformat(data["joined_at"])
+            if data.get("joined_at")
+            else None
+        )
```

Comment:
```suggestion
        self.joined_at = instance_or_none(
            datetime.fromisoformat, data.get("joined_at")
        )
```
This should work but i believe it may crate a typing issue?

Maybe should have typing `Callable[[...], T]` :thinking: